### ISSUE TITLE
Include a code snippet to function as a template for addon.json

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#6641]
### Why this would be helpful

Sometimes you want to get a head start, especially if you keep having to rewrite a structure such as an addon manifest over and over again. I know I have.

### My suggestion

I just figured out this really cool thing you can do in Visual Studio Code called Code Snippets. Basically, it's like a template, helping autocomplete frequently used snippets of code.

To make one, you specify a `*.code-snippets` file in the `.vscode` directory of the project's root directory. You name the file based on the programming language, so if I wanted to write one for JSON, I'd name it `json.code-snippets`.

To use it, you just need to have `editor.suggest.showSnippets` and IntelliSense enabled and type in the prefix of the code snippet.

You can specify where the cursor should be placed with dollar signs, and if you have multiple, you can <kbd>Tab</kbd> between them.

I made one for addon manifests (I'm still working on it):
```json
{
  "Scratch Addons manifest": {
    "prefix": "addon",
    "body": [
      "{",
      "  \"name\": \"$1\",",
      "  \"description\": \"$2\",",
      "  \"tags\": [$3],",
      "  \"credits\": [",
      "    {",
      "      \"name\": \"$4\",",
      "      \"link\": \"https://scratch.mit.edu/users/$5/\"",
      "    }",
      "  ],"
      "  \"user${6:scripts}\": [",
      "    {"
      "      \"url\": \"${7:userscript.js}\","
      "      \"matches\": [$8]"
      "    }$0",
      "  ],",
      "  \"dynamicEnable\": false,",
      "  \"dynamicDisable\": false,",
      "  \"versionAdded\": \"1.${9:0}.0\","
      "  \"enabledByDefault\": false"
      "}",
      ""
    ],
    "description": "addon.json",
  }
}
```

**Comment by Samq64** [comment]
Related: https://github.com/scratchaddons-community/create-sa-addon (I haven't tried it yet.)

**Comment by WorldLanguages** [comment]
Yeah this would be cool. There could also be a template for userscript entrypoint files.

**Comment by DNin01** [comment]
Would it be possible to add some further autocompletion such as `"info": {...}` to https://github.com/ScratchAddons/manifest-schema? Like, it'd be really cool if a JSON schema could autofill parts of notices and settings such as the name and ID. I'm always looking around for the structures of those. The less I have to do, the less I have to remember.

**Comment by WorldLanguages** [comment]
> Like, it'd be really cool if a JSON schema could autofill parts of notices and settings such as the name and ID. I'm always looking around for the structures of those.

Same

**Comment by WorldLanguages** [comment]
I never got into contributing to the addon.json schema myself, but there's probably more ways we can enhance it

**Comment by DNin01** [comment]
Also, what do you think of snippets for smaller things like settings and custom CSS variables?

**Comment by WorldLanguages** [comment]
I don't know how those can be accessed through VS Code but they probably would be a nice addition?

**Comment by DNin01** [comment]
Do we have a preferred order of manifest keys? Like, `tags` especially is all over the place across addon manifests. Where should it go? Sometimes it's below `description`, sometimes below everything else but above `dynamicEnable`, sometimes below it?

And any other order preferences like how `customCssVariables`, `settings`, `userstyles`, and `userscripts` should be ordered?

**Comment by DNin01** [comment]
> I don't know how those can be accessed through VS Code but they probably would be a nice addition?

Here's what you do:

1. Open the ScratchAddons directory/folder in VS Code
2. Make sure there's a `json.code-snippets` file in `.vscode`. You can add the one I made above.
3. Create a new file.
4. Start by typing "addon" - you'll see IntelliSense suggest the snippet. Press Enter to accept the suggestion.
5. Fill in the blanks. Press <kbd>Tab</kbd> to advance to the next cursor placement.

**Comment by WorldLanguages** [comment]
> Do we have a preferred order of manifest keys?

Why is that important? To make it easier to the eye to read manifests?
For that we should be looking into a way to move `customCssVariables` into a separate JSON file first :P

**Comment by iqnite** [comment]
Related: #5801
